main.go

```diff
@@ -110,13 +110,23 @@ func secureNetwork(testNetwork *network.Network, pace time.Duration) {
 
 	for _, peer := range testNetwork.Peers {
 		weightOfPeer := float64(testNetwork.WeightDistribution.Weight(peer.ID))
+
 		if 1000*weightOfPeer <= largestWeight {
 			continue
 		}
 
 		relevantValidators++
 
-		go startSecurityWorker(peer, time.Duration(largestWeight/weightOfPeer*float64(pace/time.Millisecond))*time.Millisecond)
+		// Weight: 100, 20, 1
+		// TPS: 1000
+		// Sleep time: 121/100000, 121/20000, 121/1000
+		// Issuing message count per second: 100000/121 + 20000/121 + 1000/121 = 1000
+
+		// Each peer should send messages according to their mana: Fix TPS for example 1000;
+		// A node with a x% of mana will issue 1000*x% messages per second
+		issuingPeriod := config.NodesTotalWeight / config.TPS / weightOfPeer
+
+		go startSecurityWorker(peer, time.Duration(issuingPeriod*float64(pace/time.Millisecond))*time.Millisecond)
```

Comment:
3 things:
1. I don't get why you divide and multiply by time.Milliseconds.. doesn't it cancel itself out? It is not so readable... Won't it be clear to write something like this?
`		go startSecurityWorker(peer, time.Duration(issuingPeriod*float64(pace))*time.Second)` (assuming we want to usr seconds becuase of `TPS` const).
2. I think `pace` is misleading because it isn't the real pace. Maybe change the name to `decelerationFactor`? This is becuase any value above 1 will deccelerate the rate of the security worker.

Reply:
Agreed. The Milliseconds unit is just to reuse the original `pace` logic and assume the simulation can run 10^9 times faster than the real simulation (we didn't multiply it by time.Second). I changed that in the new commits. [Here](https://github.com/iotaledger/multiverse-simulation/pull/2/commits/86d0383f01f1733832b093de1599424bc71752be) and [here](https://github.com/iotaledger/multiverse-simulation/pull/2/commits/d9c88949fb50094abd4538006307539125851827).